### ISSUE TITLE
[CP] Fix path name to discover debug apk on add2app builds

### ISSUE THREAD

**Issue by burnermanx** [flutter/flutter#121942]
### issue_link

https://github.com/flutter/flutter/issues/121256

### Commit Hash

5dabe102a2581a134dbf10ffc7fc8458d9bd1c05

### Target

stable

### pr_link

https://github.com/flutter/flutter/pull/121881

### Impacted Users

Flutter devs using add2app on Android

### Impact Description

Devs cannot run an add2app project just running `flutter run` or run on Andriod Studio because will not find apk for installation and will fail.

### Workaround

Create a symbolic link for apk new location, but will need to do this again if you clean the project:
>Same here, with a symbolic link it's an ok workaround
ln -s build/host/outputs/apk/debug/app-debug.apk build/host/outputs/apk/app-debug.apk

### Risk

low

### Test Coverage

yes

### Validation Steps

Basically creating an add2app project and try to run using `flutter run`

**Comment by itsjustkevin** [comment]
@reidbaker could you take a look at this cherry-pick.

**Comment by reidbaker** [comment]
Approved the pr
